source/particle/particle.cc

```diff
@@ -26,159 +26,141 @@ namespace aspect
   {
     template <int dim>
     inline
-    BaseParticle<dim>::BaseParticle (const Point<dim> &new_loc,
-                                     const double &new_id)
+    Particle<dim>::Particle (const Point<dim> &new_loc,
+                             const types::particle_index &new_id)
       :
       location (new_loc),
       id (new_id),
-      is_local (true),
-      check_vel (true)
+      properties ()
     {
     }
 
     template <int dim>
     inline
-    BaseParticle<dim>::BaseParticle ()
+    Particle<dim>::Particle ()
       :
       location (),
-      velocity (),
       id (0),
-      is_local (true),
-      check_vel (true)
+      properties()
     {
     }
 
 
     template <int dim>
     inline
-    BaseParticle<dim>::~BaseParticle ()
+    Particle<dim>::Particle (const void *&data,
```

Comment:
Ah, this is unpleasant -- unlike in the `write` and `read` functions, you can't here return the updated `data` pointer because a constructor has no return value :-( Make sure it is properly documented that the constructor alters the first argument.